CRM/Mailing/Info.php

```diff
@@ -92,33 +92,31 @@ public function getAngularModules() {
       ),
     );
 
-    $config = CRM_Core_Config::singleton();
     $session = CRM_Core_Session::singleton();
     $contactID = $session->get('userID');
-    $civiMails = civicrm_api3('Mailing', 'get', array());
-    $campNames = civicrm_api3('Campaign', 'get', array());
-    $mailingabNames = civicrm_api3('MailingAB', 'get', array());
-    $mailStatus = civicrm_api3('MailingJob', 'get', array());
-    $groupNames = civicrm_api3('Group', 'get', array());
-    $headerfooterList = civicrm_api3('MailingComponent', 'get', array());
+
+    $params = array('options' => array('limit' => 0));
+    $civiMails = civicrm_api3('Mailing', 'get', $params);
```

Comment:
For most of these entities, I think the balance of "likely record size" x "likely record count" makes it reasonable to prefetch. This one worries me a little. The basic options seem to be:

 * Set 'return'=>array('id','name','scheduled_date')
 * Rewrite crmMailingRecipients to use AJAX instead of prefetch